### PR TITLE
Feature/public release

### PR DESCRIPTION
**Pull request by sakshamphul**
## Description

PR to shape LOH towards public release

Link to cavatica app: https://cavatica.sbgenomics.com/u/d3b-bixu/tumor-loh-dev/apps/loh/7

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Test A: https://cavatica.sbgenomics.com/u/d3b-bixu/tumor-loh-dev/tasks/42627cae-61b8-49bd-b50a-6b606b89c080/
- [X] Test B: https://cavatica.sbgenomics.com/u/d3b-bixu/tumor-loh-dev/tasks/0d567a50-decc-45bf-89e4-414f881319b5/

**Test Configuration**:
* Environment:
* Test files:

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings
- [ ] I have committed any related changes to the PR

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,35 +1,63 @@
-# D3b: Loss of Heterozygosity (LOH)
+# Kids First Loss of Heterozygosity (LOH)
 
-This CWL workflow assesses the loss of heterozygosity(LOH) in the tumor for rare germline variants (gnomad_3_1_1_AF_popmax < 0.01 or gnomad_3_1_1_AF_popmax == '.'). This workflow has been deployed on [cavatica](https://cavatica.sbgenomics.com/u/d3b-bixu/tumor-loh-dev/apps/Loss_of_Heterozygosity). It was tested with a [proband only run](https://cavatica.sbgenomics.com/u/d3b-bixu/tumor-loh-dev/tasks/522d128a-2195-4c9c-8339-1709da16821d/) and a [trio run](https://cavatica.sbgenomics.com/u/d3b-bixu/tumor-loh-dev/tasks/d7f6b667-35ef-46a7-a666-970a78ef3175/)
+![data service logo](https://github.com/d3b-center/d3b-research-workflows/raw/master/doc/kfdrc-logo-sm.png)
 
-### Repo Description
+The Kids First Loss of Heterozygosity (aka LOH) is a CWL workflow that assesses the loss of heterozygosity in the tumor for rare germline calls filtered by gnomad_3_1_1_AF_popmax (typically < 0.01) or when gnomad_3_1_1_AF_popmax is not defined. This workflow is designed to analyze LOH for family trios as well as multiple proband tumor samples.
```
Comment (sickler-alex) [inline README.md:5]: I think this should be "he Kids First Loss of Heterozygosity Workflow (aka LOH) is a CWL workflow" ...

File: workflow/run_LOH_app.cwl [hunk 2]
```diff
@@ -3,7 +3,57 @@
...
+    frequency: { doc: provide popmax cutoff for rare germline variants, type: 'float?', default: 0.01 }
+    # Optional
+    ram_germline: {  doc: Provide ram (in GB) based on the size of vcf,type: 'int?', default: 8}
+    # Required for family trios otherwise not required
+    peddy_file: { doc: provide ped file for the trio, type: 'File?' }
+  Tumor tool
+    # Required
+    participant_id: { doc: provide participant id for this run, type: string }
+    bamscrams: { doc: tumor input file in cram or bam format with their index file, type: 'File[]' , secondaryFiles: [ { pattern: ".crai", required: false }, { pattern: ".bai", required: false } ] }
+    reference: { doc: human reference in fasta format with index file, type: File,secondaryFiles: [ .fai ],"sbg:suggestedValue": {class: File, path: 60639014357c3a53540ca7a3, name: Homo_sapiens_assembly38.fasta} }
```
Comment (migbro) [inline workflow/run_LOH_app.cwl:42]: One small thing I think to round this out would be to add also the secondary file as a suggested input, so:
`reference: { doc: human reference in fasta format with index file, type: File,secondaryFiles: [ .fai ],"sbg:suggestedValue": {class: File, path: 60639014357c3a53540ca7a3, name: Homo_sapiens_assembly38.fasta, secondaryFiles: [{class: File, path: 60639016357c3a53540ca7af, name: Homo_sapiens_assembly38.fasta.fai}]} }`
Just double check that I have the brackets properly placed

File: README.md [hunk 3]
```diff
@@ -1,35 +1,63 @@
...
+Germline tool filters germline annotations to retain variants based on gnomad_3_1_1_AF_popmax (typically < 0.01) or when gnomad_3_1_1_AF_popmax is not defined. It requires vcf file, proband sample id, ram as required inputs and peddy file as optional input which is required for family trios. It outputs variant information such as gene, chr, start, stop, ref/alt alleles, ref/alt allele depths, variant allele frequency and list of coordinates that will be an input to tumor tool.
 
+#### Tumor Tool
+Tumor tool search in paired proband tumor sample for aligned reads in the regions where rare variants from the germline tool exists and exact allele/reference count, allele/reference depth and calculate variant allele frequency(VAF). Tumor tool have the capability to search in multiple tumor samples for proband and if applicable, parental and maternal tumor samples. To exact reads from the bam/cram files, this tool utilizes [bam-readcount](https://github.com/genome/bam-readcount) and wraps it with python script to shape the output in a tabular format.
```
Comment (sickler-alex) [inline README.md:22]:
```suggestion
Search in paired proband tumor sample for aligned reads in the regions where rare variants from the germline exist and exact allele/reference count, allele/reference depth, and calculate variant allele frequency(VAF). Tumor tool has the capability to search in multiple tumor samples for proband and if applicable, paternal and maternal tumor samples. To extract reads from the bam/cram files, this tool utilizes [bam-readcount](https://github.com/genome/bam-readcount) and wraps it with a python script to shape the output in a tabular format. 
```

**Review by sickler-alex — CHANGES_REQUESTED** [review]

File: workflow/run_LOH_app.cwl [hunk 4]
```diff
@@ -3,7 +3,57 @@
 cwlVersion: v1.2
 class: Workflow
 id: run_LOH_app
```
Comment (migbro) [inline workflow/run_LOH_app.cwl:5]: I would probably change to `kf-loss-of-heterozygosity-wf` to better fit others we have loaded in pub apps, or also add `preprocessing` per my review comment if deemed aproraite

**Review by migbro — CHANGES_REQUESTED** [review]
One small change to add the secondary file also as the suggested input. Lastly, @jharenza just to clarify, is this actually an LOH wf in that in the end, these are LOH candidates, or more of a prepossessing wf where you use this info to determine LOH? If the latter, I'd recommend @sakshamphul to replace wf reference to LOH to Preprocessing LOH as a last update

File: workflow/run_LOH_app.cwl (hunk 2 above)
Comment (jharenza) [reply workflow/run_LOH_app.cwl:42]: this is a preprocessing workflow yes

**Review by migbro — APPROVED** [review]
looks good! I'll add you to apps pub project as well

**Review by sickler-alex — APPROVED** [review]

**Pull request merged**